src/SmartTransactionsController.ts

```diff
@@ -590,9 +606,12 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
         nonce: incrementNonceInHex(unsignedApprovalTransactionWithNonce.nonce),
       };
     } else {
-      unsignedTradeTransactionWithNonce = await this.addNonceToTransaction(
-        tradeTx,
-      );
+      unsignedTradeTransactionWithNonce = tradeTx;
+      if (!unsignedTradeTransactionWithNonce.nonce) {
```

Comment:
This reads a little odd to me. Var name is `withNonce` but it might not exist on the object.